tests/collection-access.ts

```diff
@@ -38,14 +59,25 @@ describe('Map', () => {
   test('get with value', () => {
     expect(map.get('a')).toBe(1)
     expect(map.get('a', true)).toMatchObject({ value: 1 })
-    expect(map.get('b').toJSON()).toMatchObject({ c: 3, d: 4 })
+    const subMap = map.get('b')
+    if (!isMap<string, number>(subMap)) {
+      throw new Error('Expected subMap to be a Map')
+    }
+    expect(subMap.toJSON()).toMatchObject({
+      c: 3,
+      d: 4
+    })
```

Comment:
```suggestion
    if (isMap<string, number>(subMap)) {
      expect(subMap.toJSON()).toMatchObject({ c: 3, d: 4 })
    } else {
      throw new Error('Expected subMap to be a Map')
    }
```

I like this style of assertion + cast. For readability, I would prefer the throws to come from explicit rather than implicit branches, though. Same goes for later tests as well.